### DISCUSSION TITLE
Failing to use GroupFilter, generated exception

### DISCUSSION THREAD

**Discussion by IvanNavarroSommerz** [codeigniter4/shield#930] [Q&A]
- I defined this route in APP/Config/routes.php:

`$routes->get('/admin/(:segment)', 'AdminCtl::index/$1', ['filter' => 'hasAccess:superadmin,admin']);`

- and defined an alias at APP/Config/filters.php

```
 _public array $aliases = [
  ...
  'hasAccess'   => [
                             GroupFilter::class,
         ],
 ]_

``` 

- Using session authenticator.

- At line 688 in Session.php,  function getSessionUserInfo() breaks, generating an exception: 

`_Call to undefined function CodeIgniter\Shield\Authentication\Authenticators\setting() _`

I tried a few things but with no success, i believe I'm missing something. 
This same call at getSessionUserInfo() succeeds in every other context, except when called via the filter path.

I installed both shield and settings libraries manually, not via composer and I'm not sure if it makes any difference. 
In APP/Config/Autoload.php, I have:
```
_public $psr4 = [
        ...
        'CodeIgniter\\Settings' => APPPATH . 'ThirdParty/settings/src',
        'CodeIgniter\\Shield' => APPPATH . 'ThirdParty/shield/src', 

    ];_

```
Any comments mostly appreciated!
Cheers

**Comment by datamweb** [comment]
> * and defined an alias at APP/Config/filters.php

Before anything, you should know that all shield filters are loaded by default. Therefore, there is no need to add filters to file **APP/Config/filters.php**.

```php
$routes->get('/admin/(:segment)', 'AdminCtl::index/$1', ['filter' => 'session:superadmin,admin']);
```

> * At line 688 in Session.php,  function getSessionUserInfo() breaks, generating an exception:
> 
> `_Call to undefined function CodeIgniter\Shield\Authentication\Authenticators\setting() _`

Please follow the instructions in the document below to upgrade properly.

https://github.com/codeigniter4/shield/blob/develop/UPGRADING.md

**Reply by IvanNavarroSommerz** [reply]
Beautiful thanks, I ll do.

**Comment by datamweb** [comment]
> * and defined an alias at APP/Config/filters.php

@Kenjis I highlighted this in the review, and you moved the note up. But in the last few days, I noticed that many users are reloading the filters by mistake. What should be changed so that the documentation is clear on this matter?

https://github.com/remunata/techporia23/pull/1
https://github.com/Memurame/easy-customer-management/pull/4
and ...

**Reply by IvanNavarroSommerz** [reply]
In my case, I'm fairly new to this, since there's no mentioning on applying the filter in the route itself, i resorted to codeigniter's docs on applying filters to routes and went through that path forgetting about the note at shield's doc mentioning that filters are loaded by default. Maybe a little example will help.

**Reply by kenjis** [reply]
I added a few explanations. See #931
